File: two_pointers_pattern/sum_equals_zero.rb


def sum_zero(arr)
  left = 0
  right = arr.length - 1

  while left < right do
    sum = arr[left] + arr[right]
    return [arr[left], arr[right]] if sum == 0
    sum > 0 ? right -= 1 : left+= 1
  end
end

File: two_pointers/sumEqualsZero.js
// Write a function that takes a sorted array and returns the pair that sum is equal to 0. 
// Otherwise, return undefined.

function sumZero(arr){
	let left = 0;
	let right = arr.length - 1;
	while(left < right){
		let sum = arr[left] + arr[right];
		if(sum === 0){
			return [arr[left], arr[right]];
		} else if(sum > 0){
			right--;
		} else {
			left++;
		}
	}
}
